jscomp/ml/translcore.ml

```diff
@@ -802,6 +809,33 @@ and transl_exp0 (e : Typedtree.expression) : Lambda.lambda =
           | _ ->
               wrap (Lprim (prim, argl, e.exp_loc))
               ))
+  | Texp_apply
+      ( {
+          exp_desc =
+            Texp_ident
+              ( Pdot (Pdot (Pident module_name, "Dict", _), "fromArray", _),
+                _,
+                _ );
+        },
+        [(_lbl, Some {exp_desc = Texp_array args})] )
+    when Ident.name module_name = "Js" && args_eligible_for_dict_inlining args ->
```

Comment:
I think there's some code pattern that stores the path to specific functions ahead of time so one does not have to guess from name here.
It should be true eg for all the runtime functions CC @cknitt

Reply:
@cristianoc there's a pattern like that in the code base already? Do you know where?

Reply:
So it looks like all such things are more principled. E.g. `import` is a primitive (with `#`):
```
external import: 'a => promise<'a> = "#import"
```

What's fishy here is a hard-coded path to something that exists both as `Js.Dict` and some other path in Core which is not in the compiler yet.

So if this goes, then there has to be some clean up later, to avoid bugs etc.

Reply:
Would you suggest to do this the same way as with await then? There's an experiment doing something similar that was removed here, check the earlier commits in this branch if you're interested.

Reply:
Ping @cristianoc

Reply:
Up to preference. One is quicker. The other is safer.